[2.0.r1] cp_headers: Add ADSP sleep monitor header

bartcubbins opened the pull request:
This header is required for the new PAL (Platform Audio Layer).

bartcubbins committed:
cp_headers: Add ADSP sleep monitor header

This header is required for the new PAL (Platform Audio Layer).

Signed-off-by: Pavel Dubrova <pashadubrova@gmail.com>

jerpelea commented:
please include the header

jerpelea reviewed (changes requested):
please include the header file

bartcubbins commented:
Sure. Please leave this open for a while, I just found that the header list is out of sync with the current kernel version. I will append this PR soon

bartcubbins committed:
cp_headers: Update msm_media_info.h path

msm_media_info.h header is in the media subdirectory
and the output header should be there too.

warning: 'kernel/sony/msm-5.15/common-headers/original-kernel-headers/msm_media_info.h' does not exist

Signed-off-by: Pavel Dubrova <pashadubrova@gmail.com>

bartcubbins committed:
cp_headers: Sync headers with the current kernel version

Remove the following non-existent headers:
  - msm_audio_calibration.h
  - msm_audio_wmapro.h
  - msm_audio_amrwb.h
  - msm_audio_amrwbplus.h
  - msm_audio_wma.h
  - msm_audio_aac.h

Remove internal header wcd9xxx_registers.h which is not
used by any external module.

Add new audio_slimslave.h header.

Signed-off-by: Pavel Dubrova <pashadubrova@gmail.com>

bartcubbins committed:
cp_headers: Sort headers alphabetically

This is just a cosmetic change that sorts the headers
alphabetically and removes duplicates.

Signed-off-by: Pavel Dubrova <pashadubrova@gmail.com>

bartcubbins commented:
PR has been updated, but we still have remnants headers from previous kernel versions such as `msm_ion.h`. The APIs they represent have been removed from the kernel, so having these headers here is very dangerous

bartcubbins commented:
PR has been updated, but we still have remnant headers from previous kernel versions such as `msm_ion.h`. The APIs they represent have been removed from the kernel, so having these headers here is very dangerous

jerpelea commented:
please remove the old deprecated headers in a separate commit

bartcubbins commented:
Force pushed only because `wcd-dsp-glink.h`  header was returned to the list. It was accidentally lost in the sync commit.
CLO (ex. CAF) still supplies UAPI msm_ion headers for backwards compatibility[1]. So we are good to go. Everything is up to date.
[1] https://github.com/sonyxperiadev/kernel/commit/1c173b1bb323a5e4cf8f7d0184c6bc9a5bee4909
